### ISSUE TITLE
Nuclei dataset preprocessing

### ISSUE THREAD

**Issue by nikku1234** [tommy-qichang/AsynDGAN#9]
Can you explain the preprocessing done on the nuclei dataset? And how should I train and test the code for nuclei dataset.

**Comment by huiqu18** [comment]
@nikku1234 For preprocessing, all images in the nuclei dataset were color normalized using Reinhard's method ([44] in the paper). In the training set, each image of 1000x1000 was split into 16 image patches of 286x286. The same process was done for labels. The image and label patches were the input to GAN (load size 286, crop size 256). Then the training set was split into four subsets according to four different organs.

In the `data/nuclei_dataset.py`, `train_all.h5` is the whole training set, `train_breast.h5`, `train_kidney.h5`, `train_liver.h5`, `train_prostate.h5` are the four subsets.

To train the GAN, run
`python train.py --dataroot <path-to-nuclei-data> --name nuclei_asyndgan --model asyndgan --netG resnet_9blocks --direction AtoB --lambda_L1 100 --dataset_mode nuclei_split --pool_size 0 --gpu_ids 0 --batch_size 2 --num_threads 0`

To generate the synthetic images using the trained model, run
`python save_syn_nuclei.py --dataroot <path-to-nuclei-data> --name nuclei_asyndgan --model pix2pix --netG resnet_9blocks --direction AtoB --dataset_mode nuclei --epoch 400 --results_dir results/nuclei_asyndgan`

Note that the generator of our AsynDGAN is the same as that of pix2pix, that's why we simply used pix2pix for testing.

**Comment by nikku1234** [comment]
Oh awesome! Thanks for the detailed explanation.
